timemachine/fe/bar.py

```diff
@@ -126,43 +174,34 @@ def bootstrap_bar(w_F, w_R, n_bootstrap=1000, timeout=10) -> Tuple[float, NDArra
     * TODO[deboggle] -- upgrade from pymbar3 to pymbar4 and remove this
     * TODO[performance] -- multiprocessing, if needed?
     """
-    full_bar_result = pymbar.BAR(w_F, w_R, compute_uncertainty=False)
+    u_kn, N_k = ukln_to_ukn(u_kln)
+    mbar = pymbar.MBAR(u_kn, N_k)
 
-    n_F, n_R = len(w_F), len(w_R)
+    full_bar_result = mbar.getFreeEnergyDifferences(compute_uncertainty=False)[0][0, 1]
 
-    bootstrap_samples = []
+    _, _, n = u_kln.shape
 
-    t0 = time()
+    bootstrap_samples = []
 
     seed = 2022
     rng = np.random.default_rng(seed)
 
     for _ in range(n_bootstrap):
-        elapsed_time = time() - t0
-        if elapsed_time > timeout:
-            break
-
-        w_F_sample = rng.choice(w_F, size=(n_F,), replace=True)
-        w_R_sample = rng.choice(w_R, size=(n_R,), replace=True)
-
-        bar_result = pymbar.BAR(
-            w_F=w_F_sample,
-            w_R=w_R_sample,
-            DeltaF=full_bar_result,  # warm start
-            compute_uncertainty=False,
-            relative_tolerance=1e-6,  # reduce cost
-        )
+        u_kln_sample = rng.choice(u_kln, size=(n,), replace=True, axis=2)
 
+        bar_result = df_from_u_kln(
+            u_kln_sample,
+            initial_f_k=mbar.f_k,  # warm start
```

Comment:
q: would it make sense to add bootstrap_maximum_iterations to signature of bootstrap_bar, then forward it here?

Reply:
Unsure whether it would be useful to be able to specify max iterations separately for the point estimate and the bootstrap samples, but this can probably be added later if useful.

Reply:
>Unsure whether it would be useful to be able to specify max iterations separately for the point estimate and the bootstrap samples, but this can probably be added later if useful.

Separately makes sense to me (to control expense), but can be added later if needed.